examples/vision/image-classification/pets/quantize_vit_model.py

```diff
@@ -0,0 +1,132 @@
+import argparse
+import time
+from tempfile import NamedTemporaryFile
+
+import torch
+import torch.nn.functional as F
+from accelerate import init_empty_weights
+from datasets import load_dataset
+from safetensors.torch import load_file, save_file
+from transformers import (
+    ViTForImageClassification,
+    ViTImageProcessor,
+    ViTConfig,
+)
+
+from optimum.quanto import (
+    Calibration,
+    QTensor,
+    freeze,
+    qfloat8,
+    qint4,
+    qint8,
+    quantization_map,
+    quantize,
+    requantize,
+)
+
+
+def test(model, device, test_loader):
+    model.to(device)
+    model.eval()
+    test_loss = 0
+    correct = 0
+    with torch.no_grad():
+        start = time.time()
+        for batch in test_loader:
+            data, target = batch["pixel_values"], batch["labels"]
+            data, target = data.to(device), target.to(device)
+            output = model(data).logits
+            # print("*****I am after output", output)
+            if isinstance(output, QTensor):
+                output = output.dequantize()
+            test_loss += F.nll_loss(output, target, reduction="sum").item()  # sum up batch loss
+            pred = output.argmax(dim=1, keepdim=True)  # get the index of the max log-probability
+            correct += pred.eq(target.view_as(pred)).sum().item()
+        end = time.time()
+
+    test_loss /= len(test_loader.dataset)
+
+    print(
+        "\nTest set evaluated in {:.2f} s: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n".format(
+            end - start, test_loss, correct, len(test_loader.dataset), 100.0 * correct / len(test_loader.dataset)
+        )
+    )
+
+
+def keyword_to_itype(k):
+    return {"none": None, "int4": qint4, "int8": qint8, "float8": qfloat8}[k]
+
+
+def main():
+    parser = argparse.ArgumentParser(description="ViT PETS Example")
+    parser.add_argument("--model", type=str, default="super-j/vit-base-pets")
+    parser.add_argument("--device", type=str, default=None, help="The device to use for evaluation.")
+    parser.add_argument("--weights", type=str, default="int8", choices=["int4", "int8", "float8"])
+    parser.add_argument("--activations", type=str, default="int8", choices=["none", "int8", "float8"])
+    args = parser.parse_args()
+
+    # torch.manual_seed(args.seed)
+
```

Comment:
```suggestion
```

Reply:
removed it in https://github.com/huggingface/optimum-quanto/pull/308/commits/bd0332fbaac1d7f580a1be7cadb0c972f6779536